storage/innobase/handler/handler0alter.cc

```diff
@@ -6119,20 +6119,20 @@ static bool innobase_instant_try(
 	/* Convert the table to the instant ALTER TABLE format. */
 	mtr.commit();
 	mtr.start();
+	mtr_x_lock_index(index, &mtr);
 	index->set_modified(mtr);
 	if (buf_block_t* root = btr_root_block_get(index, RW_SX_LATCH, &mtr)) {
 		if (fil_page_get_type(root->frame) != FIL_PAGE_INDEX) {
 			DBUG_ASSERT(!"wrong page type");
 			goto err_exit;
 		}
 
-		btr_set_instant(root, *index, &mtr);
-		mtr.commit();
-		mtr.start();
-		index->set_modified(mtr);
 		err = row_ins_clust_index_entry_low(
 			BTR_NO_LOCKING_FLAG, BTR_MODIFY_TREE, index,
 			index->n_uniq, entry, 0, thr);
+		if (err == DB_SUCCESS) {
+			btr_set_instant(root, *index, &mtr);
+		}
```

Comment:
Sorry, I realized (after testing this on 10.6) that the patch is incorrect. We must pass `mtr` to `row_ins_clust_index_entry_low()`.

Reply:
Addressed in 54b1ff7531058eca238e939eaa1c9a397a672205. I am working on a separate 10.6 version of this fix.